File: Client/src/dtos/postDtos.ts
export interface CreatePostDto {
  caption?: string | null;
}

export interface UpdatePostCaptionDto {
  caption?: string | null;
}

export interface PostDto {
  id: string;
  caption?: string;
  viewCount: number;
  likeCount: number;
  commentsCount: number;
  mediaId: string;
  postOwnerId: string;
  postOwnerUserName: string;
  isYours: boolean;
  didYouLiked: boolean;
  postedAt: string;
}

export interface PostListItemDto {
  id: string;
  viewCount: number;
}

export interface CreatePostResponseDto {
  id: string;
}

File: Server/Socialgram/DTOs/PostDtos.cs

namespace Socialgram.DTOs
{
    public record class CreatePostDto(string? Caption);

    public record class PostDto(Guid Id,
                                string? Caption,
                                int ViewCount,
                                int LikeCount,
                                int CommentsCount,
                                Guid MediaId,
                                Guid PostOwnerId,
                                string PostOwnerUserName,
                                bool IsYours,
                                bool DidYouLiked,
                                DateTime PostedAt);

    public record class UpdatePostCaptionDto(string? Caption);

    public record class PostListItemDto(Guid Id,
                                        int ViewCount);
}
